pulsar/internal/connection.go

```diff
@@ -309,7 +309,13 @@ func (c *connection) WriteData(data []byte) {
 
 func (c *connection) internalWriteData(data []byte) {
 	c.log.Debug("Write data: ", len(data))
-	if _, err := c.cnx.Write(data); err != nil {
+	err := c.cnx.SetWriteDeadline(time.Now().Add(time.Second * 15))
```

Comment:
Should this be set once per connection or once per write? Also, this should be a variable and not hardcoded so we can make it configurable in the future if needed.

Reply:
Makes sense. Done.